From 11e7f153a7f24d0123bf0123472c11f36b7dcc3c Mon Sep 17 00:00:00 2001
From: Jason Yu <xuzj03@gmail.com>
Date: Fri, 30 Nov 2018 00:41:26 +0800
Subject: [PATCH 1/4] Open flag riscv_slow_unaligned_access

 - The RV32M1 does not support 16-bit unaligned write.
   For a packed structure, the store half-word should
   be split into two store byte.
---
 gcc/config/riscv/riscv.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gcc/config/riscv/riscv.c b/gcc/config/riscv/riscv.c
index 5fd9835c78e..cc80d4590b8 100644
--- a/gcc/config/riscv/riscv.c
+++ b/gcc/config/riscv/riscv.c
@@ -4992,11 +4992,13 @@ riscv_option_override (void)
   /* Use -mtune's setting for slow_unaligned_access, even when optimizing
      for size.  For architectures that trap and emulate unaligned accesses,
      the performance cost is too great, even for -Os.  */
-/*
+
+#if 1
   riscv_slow_unaligned_access = (cpu->tune_info->slow_unaligned_access
 				 || TARGET_STRICT_ALIGN);
-*/
+#else
   riscv_slow_unaligned_access = 0;
+#endif
   /* If the user hasn't specified a branch cost, use the processor's
      default.  */
   if (riscv_branch_cost == 0)
-- 
2.17.1

